src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/dto/TestSuiteDTO.java

```diff
@@ -20,4 +22,14 @@ public TestSuiteDTO(String name, double time, int errors, int skipped, int failu
         this.tests = tests;
         this.testCases = testCases;
     }
+
+    @Override
+    public List<TestCaseDTOInterface> getFailedTests() {
+        return testCases.stream().filter(testCase -> !testCase.isSuccessful()).map(testCase -> (TestCaseDTOInterface) testCase).toList();
```

Comment:
I’ve seen this in a few places. The explicit lambda could be avoided in those cases. The current version is perfectly fine, though, if you prefer.

```suggestion
        return testCases.stream().filter(testCase -> !testCase.isSuccessful()).map(TestCaseDTOInterface.class::cast).toList();
```